ibtk/include/ibtk/IBTKInit.h

```diff
@@ -0,0 +1,110 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2014 - 2019 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+#ifndef included_IBTKInit
+#define included_IBTKInit
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <IBTK_config.h>
+
+#include "ibtk/IBTK_MPI.h"
+
+#include "SAMRAI_config.h"
+#include "tbox/SAMRAIManager.h"
+#include "tbox/SAMRAI_MPI.h"
+
+#include <petscsys.h>
+#ifdef IBTK_HAVE_LIBMESH
+#include "libmesh/libmesh.h"
+#include "libmesh/reference_counter.h"
+#endif
+
+namespace IBTK
+{
+/**
+ * @brief Initialization for IBAMR programs.
+ *
+ * The singleton IBTKInit class handles the initializations for PETSc, LibMesh, and SAMRAI. This object should be
+ * created using the initialize() function at the start of the main() function. The destruction of the object correctly
+ * closes the libraries.
+ *
+ */
+
+class IBTKInit
+{
+public:
+    /**
+     * Constructor for IBTKInit. Initializes libraries and sets the SAMRAI world communicator. Attempts to create a
+     * second IBTKInit object will result in a run time error.
+     */
+    IBTKInit(int argc,
+             char** argv,
+             IBTK_MPI::comm communicator = MPI_COMM_WORLD,
```

Comment:
Lets just use `MPI_comm` here instead of the alias. That makes it a bit clearer that `IBTK_MPI::comm` isn't some C++ wrapper class.

Reply:
Fixed.